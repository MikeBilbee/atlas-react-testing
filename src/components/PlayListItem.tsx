// PlayListItem.tsx
import React from 'react';

interface PlayListItemProps {
	title: string;
	artist: string;
	duration: string;
}

const PlayListItem: React.FC<PlayListItemProps> = ({ title, artist, duration }) => {
	
	return (
		<div className="flex items-center justify-between py-1 px-5">
			<div>
				<h3 className="text-lg font-medium dark:text-gray-400">{title}</h3>
				<p className="dark:text-gray-500 text-sm">{artist}</p>
			</div>
			<p className="dark:text-gray-500 text-sm">{duration}</p>
		</div>
	);
};

export default PlayListItem;