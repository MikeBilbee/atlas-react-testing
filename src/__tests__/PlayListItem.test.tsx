//PlayListItem.test.tsx
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import PlayListItem from '../components/PlayListItem';

describe('PlayListItem component', () => {
	it('renders with song details', () => {
		const song = {
			title: 'Sample Song',
			artist: 'Test Artist',
			duration: '3:45',
		};
		const { container } = render(<PlayListItem {...song} />);
		expect(container).toMatchSnapshot();
	});
});