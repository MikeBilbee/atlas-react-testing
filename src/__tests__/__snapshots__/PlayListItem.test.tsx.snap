// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem component > renders with song details 1`] = `
<div>
  <div
    class="flex items-center justify-between py-1 px-5"
  >
    <div>
      <h3
        class="text-lg font-medium dark:text-gray-400"
      >
        Sample Song
      </h3>
      <p
        class="dark:text-gray-500 text-sm"
      >
        Test Artist
      </p>
    </div>
    <p
      class="dark:text-gray-500 text-sm"
    >
      3:45
    </p>
  </div>
</div>
`;
